cbpv.zy

```diff
@@ -1,69 +1,43 @@
 codata Monad (M: VType -> CType) where
   | .return(): forall (A: VType) . A -> M A
   | .bind(): forall (A: VType) (A': VType) . U (M A) -> U (A -> M A') -> M A'
+  | .fail(): forall (A: VType) . Exception -> M A
+  // | .print(): String -> M Unit
+end
+
+codata AlgebraValue (M: VType -> CType) (B: CType) where
+  | .bindA(): U (M Value) -> U (Value -> B) -> B
```

Comment:
Because if we keep `B` to be any `CType` kinded type, the polymorphic value can't fit into all possible `B`s; instead, I suggest that we replace `B: CType` with `N: VType -> CType` to allow `N` to be aware of `A`, effectively turning `Algebra` into:

```
codata Algebra (M: VType -> CType) (N: VType -> CType) where
  | .bindA(): forall (A: VType) . U (M A) -> U (A -> N A) -> N A
end
```

At this point, I'm a bit hesitant. It seems that `Algebra` has to be built on top of a monad `M` and some `N` that takes a `VType` so that it can "know what's going on", which is trapping me for days. I'm not sure whether this extra condition attached to `B` is well-justified, but that's the best I can do for now..

Other than that, the polymorphic version is now updated.

Reply:
By the way, both the version that the conversation is happening upon and the polymorphic one are running on Zydeco right now.